src/HealthGPS/kevin_hall_model.cpp

```diff
@@ -114,6 +139,89 @@ void KevinHallModel::update_risk_factors(RuntimeContext &context) {
     }
 }
 
+void KevinHallModel::initialise_sector(RuntimeContext &context, Person &person) const {
+
+    // Get rural prevalence for age group and sex.
+    double prevalence;
+    if (person.age < 18) {
+        prevalence = rural_prevalence_.at("Under18"_id).at(person.gender);
+    } else {
+        prevalence = rural_prevalence_.at("Over18"_id).at(person.gender);
+    }
+
+    // Sample the person's sector.
+    double rand = context.random().next_double();
+    auto sector = rand < prevalence ? core::Sector::rural : core::Sector::urban;
+    person.sector = sector;
+}
+
+void KevinHallModel::update_sector(RuntimeContext &context, Person &person) const {
+
+    // Only update rural sector 18 year olds.
```

Comment:
How come?

Reply:
Also, why are you looking at people who are *exactly* 18, rather than over 18?

Reply:
Ohhh I see. You're assuming that migration to the city always happens at age 18. I think I remember Ali talking about this.